roles/lmod/tasks/main.yml

```diff
@@ -12,66 +25,98 @@
   action: yum name=lmod state=latest
   when: ansible_os_family == "RedHat"
 
+# see: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=891541
+# lmod does not create posix.so in system libraries sir
+# fix: symlink posix_c.so into posix.so
+- name: "bugfix for lomd posix_c"
+  become: yes
+  file:
+    src: "{{ lmod_sys_lib_dir }}/{{ item }}/posix_c.so"
+    dest: "{{ lmod_sys_lib_dir }}/{{ item }}/posix.so"
+    state: link
+    force: yes
+  with_items:
+    - 5.1
+    - 5.2
+    - 5.3
+  when: ansible_os_family == "Debian"
+  tags:
+    - bugfix
+
+# see: https://bugs.launchpad.net/ubuntu/+source/lmod/+bug/1850237
+# lmod has incorrect SYS_LUA_CPATH on arm64
+# fix: sed -i 's/x86_64/aarch64/g' /usr/share/lmod/lmod/libexec/lmod
+- name: "bugfix for sys_lua_path: find lua config files"
+  become: yes
+  find:
+    paths: "/usr/share/lmod/lmod/libexec"
+    patterns: "*"
+    recurse: yes
+  register: lua_config
+  tags:
+    - bugfix
+
+- name: "bugfix for sys_lua_path: set the right arch"
+  become: yes
+  replace:
+    path: "{{ item.path }}"
+    regexp: 'x86_64-linux-gnu'
+    replace: 'aarch64-linux-gnu'
+  with_items: "{{ lua_config.files }}"
+  when: ansible_os_family == "Debian" and ansible_architecture == "aarch64"
+  tags:
+    - bugfix
+
 - name: "mkdir software path"
   file: path={{ sm_software_path }} state=directory
```

Comment:
Is `sm_software_path` defined somewhere? It should have a default set in the role

Reply:
https://github.com/NVIDIA/deepops/pull/543/commits/af22cdf54b5490ac37f2a040e71b8db90563a640